### ISSUE TITLE
T3-php8.1 issue with publishing new data

### ISSUE THREAD

**Issue by jhumann** [tripal/tripal#1471]
## System information
Drupal v7.94
PostgreSQL v15.1 (Ubuntu 15.1-1.pgdg20.04+1)
PHP v8.1.17
Tripal v7.x-3.10 (latest dev branch)

### Issue description
We are doing testing with php 8.1 on our D7/T3 sites and noticed that there was an issue when loading genome data.  I have used the T3.9 gff loader on out live site with php7.x and it worked fine.  When I loaded the same genome on the dev site under php 8.1, I did notice an increased number of warning messages when using the gff file loader, but the job appeared to complete.  The data is also present in the database.  However, when I 'publish' the new gene and mRNA data, zero records are found to publish.

#### Steps to reproduce

Load genome fasta and gff files via the Tripal data loaders and try to 'publish' the new gene and mRNA trial content.
 
#### Error messages and screenshots
stdout from drush for publish jobs:

```
2023-04-05 15:45:44: Job ID 20994.
2023-04-05 15:45:44: Calling: chado_publish_records(Array)
INFO (PUBLISH_RECORDS): There are 0 records to publish.
There are 0 records to publish.
INFO (PUBLISH_RECORDS): Successfully published 0 Gene record(s).
Successfully published 0 Gene record(s).
2023-04-05 15:45:52: Job ID 20995.
2023-04-05 15:45:52: Calling: chado_publish_records(Array)
INFO (PUBLISH_RECORDS): There are 0 records to publish.
There are 0 records to publish.
INFO (PUBLISH_RECORDS): Successfully published 0 mRNA record(s).
Successfully published 0 mRNA record(s).
```

error message

There are no obvious error messages in the drush command stdout for the publish step.  If you would like the stdout from the gff and fasta loader drush jobs, I can provide text files of that so you can see the "warnings".  I don't know if that is related to the publish issue.

**Comment by dsenalik** [comment]
I just loaded one chromosome under PHP 8.1.17 and did not see any issues
There are 5351 records to publish for "Gene" (bio_data_8)
There are 25325 records to publish for "mRNA" (bio_data_9)
There are 5230 records to publish for "Polypeptide" (bio_data_36)

Could you supply a subset of the fasta and gff3 file for testing? It might be a format issue.
And yes the warnings will be helpful even if not related.

And I tested using the version from https://github.com/tripal/tripal/pull/1369
there is a chance that has fixed something you are encountering.

**Comment by jhumann** [comment]
Alright, this was a user-error.  I forgot that I had already used those genome files for a test last summer.  As a result, the features were already published.  I have since used a different genome and it is all good now.  BTW, we have noticed that the T3.9 GFF loader does not delete the temporary cache file it creates if the job is successful.  It deletes it if the job fails.  I will watch for this in T4 and will put a request in for that huge cache file to be automatically deleted if it isn't addressed in T4.  We can deal with it for the remaining time that T3 will be around:-)
